app/task_routes.py

```diff
@@ -0,0 +1,127 @@
+from flask import Blueprint
+from app import db, SLACK_API_KEY
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from datetime import datetime
+import requests
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+# Helper Functions
+def valid_int(number, parameter_type):
+    try:
+        int(number)
+    except:
+        abort(make_response({"error": f"{parameter_type} must be an int"})), 400
+
+def get_task_from_id(task_id):
+    valid_int(task_id, "task_id")
+    return Task.query.get_or_404(task_id, description="{task not found}")
+
+#Wave 2 asc and desc
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    tasks = Task.query.all()
+    tasks_response = []
+    sort_by = request.args.get("sort")
+
+    if sort_by == "asc":
+        sorted_tasks = Task.query.order_by(Task.title.asc())
+        for task in sorted_tasks:
+            task = task.to_dict()
+            tasks_response.append(task)
+        return jsonify(tasks_response)
+
+    elif sort_by == "desc":
+        sorted_tasks = Task.query.order_by(Task.title.desc())
+        for task in sorted_tasks:
+            task = task.to_dict()
+            tasks_response.append(task)
+
+        return jsonify(tasks_response)
+
+    if not tasks:
+        return jsonify(tasks_response), 200
```

Comment:
We don't need this, because the code above will return exactly this for a no tasks
```suggestion
```